### DISCUSSION TITLE
Single hyphen argument

### DISCUSSION THREAD

**Discussion by ghost** [adishavit/argh#78] [Q&A]
Is there a way in Argh to have single hyphen arguments ? 
to reproduce certain arguments from https://www.manpagez.com/man/1/xcrun/

**Comment by adishavit** [comment]
That is already supported via the `SINGLE_DASH_IS_MULTIFLAG` mode flag. 🙂

**Reply by ghost** [reply]
that makes it so single dashes flags are interpreted as regular flags?
is it a compile time flag or run time flag?

**Reply by adishavit** [reply]
It makes it so that each letter after the single dash is interpreted as a single independent flag. 
See the README for an example.

**Reply by ghost** [reply]
I've red the README and this does not seem to be what I'm seeking for

**Comment by adishavit** [comment]
Please explain what input you want and what behavior you expect and how it
differs from the existing behavior.

Give examples here not with a link.

**Reply by ghost** [reply]
I expect single dash'ed (say, -example) arguments to be interpreted as "example"
and be able to do something such as 
```rust
struct Args {
    #[argh(switch, short = 'e', description = "example")]
    example: bool,
}
```
,
and not have -example be interpreted as -e, -x, -a, ...

TL;DR, single dashed argument to be interpreted just like double dashed ones

**Reply by a4z** [reply]
I guess this 
> -xvf will be parsed as 3 separate flags: x, v, f.

should be a flag `xvf` is set.  1 flag wit xvf as name, not 3 short flags

A lot of command line programs these days have 1-hyphen arguments

cmd `-xcf=1`  instead of `cmd --xcf=1`

I learned that mostly when switching to work on a Mac some years ago, the classical option short/long, where the single hyphen is one letter arg/flag, and 2 hyphens are a named argument, and they often come in pairs, they do not exist.

It's also the default for every go program.

if you look at the provided man page, its only `-verbose` , not `-v / --verbose`

**Reply by a4z** [reply]
since I got used to this, this would be a nice addition to argh

**Reply by adishavit** [reply]
> TL;DR, single dashed argument to be interpreted just like double dashed ones

This is the default behavior. Argh does not care how many dashes you put. 

`>>myapp -arg_one --arg_two` 

Unless explicitly specified otherwise, both will be treated identically, creating two (boolean) flags called `arg_one` and  `arg_two`.
